### PR TITLE
Fix update offset and cursor position calculation

### PR DESCRIPTION
**Pull request by ihork**
Hi, Artal! Could you please review this fix?
It fixes problem with calculation of available edit area height and scroll position. On different platforms you should just start to type symbols on new lines. On iPhone 5, for example, you will see that last few lines are placed under toolbar.
Another thing, that on ios devices when you start scrolling, you are able to move text out the view and see a white gap under the text. Update offset fixes this problem.

### CODE REVIEW

File: src/editor.html [hunk 1]
```diff
@@ -952,9 +957,9 @@
 
 				var offsetY = window.document.body.scrollTop;
 
-				var footer = $('#zss_editor_footer');
+				var footer = document.getElementById('zss_editor_footer');
```
Comment (artald) [inline src/editor.html]: why did you change from original jQuery like it's done on other places?

File: src/editor.html [hunk 2]
```diff
@@ -1696,6 +1710,10 @@
 				padding-left: 10px;
 				padding-right: 10px;
 			}
+
+			#zss_editor_footer {
+				height: 10px;
```
Comment (artald) [inline src/editor.html]: why do we need a hard-coded value for footer, and why even handle the footer if we don't use it?
worried about messing with layouts in other screens that are using this component

File: src/editor.html (hunk 1 above)
Comment (ihork) [inline src/editor.html]: I did it because I was needed offsetTop and offsetHeight in my calculations and It's easier to take it directly from node. And also I noticed that inside editor there are other places where we use document methods directly, so I decided that I will not break any code style in this context.

File: src/editor.html (hunk 2 above)
Comment (ihork) [inline src/editor.html]: I used footer as a default gap between last line in editor and keyboard. It gives a better UI. I use its height in updateOffset function. DYT we should specify its height via props?

Comment (artald) [inline src/editor.html]: Since this component is supposed to be used in various places we don't want to break the functionality. Consider a case for example that this editor is not full screen but is just showing a few lines, or any other use-case.

If you can, please specify this as a prop like you suggested and the default would remain the same as now (0).

Comment (ihork) [inline src/editor.html]: Great, I agree with you. Will do it, thanks!
